lib/Rex/Fork/Task.pm

```diff
@@ -56,7 +56,18 @@ sub start {
       exit();
     }
     else {
-      &$func($self);
+      eval { &$func($self) };
+      my $exit_code = $@ ? ($? || 1) : 0;
+      push @PROCESS_LIST, $exit_code;
```

Comment:
If tasks fail, the exit value of rex should reflect that.  That doesn't currently happen.  This fixes the problem.